catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -455,9 +466,17 @@ private Path getTempContentItemDir(String requestId, URI contentUri) {
     private Path getContentItemDir(URI contentUri) {
         List<String> pathParts = getContentFilePathParts(contentUri.getSchemeSpecificPart(),
                 contentUri.getFragment());
-
-        return Paths.get(baseContentDirectory.toAbsolutePath()
-                .toString(), pathParts.toArray(new String[pathParts.size()]));
+        try {
+            return Paths.get(baseContentDirectory.toAbsolutePath()
```

Comment:
Made it so FileSystemStorageProvider Automatically canonicalizes the path when the `baseContentDirectory` is set so everything in here can just use that path as is and it should be :100: (as opposed to everyone repeating the same work every time)